lms/djangoapps/teams/static/teams/js/views/team_profile.js

```diff
@@ -43,14 +43,14 @@
                     discussionTopicID = this.model.get('discussion_topic_id'),
                     isMember = TeamUtils.isUserMemberOfTeam(memberships, this.context.userInfo.username),
                     isAdminOrStaff = this.context.userInfo.privileged || this.context.userInfo.staff,
-                    isInstructorManagedTopic = TeamUtils.isInstructorManagedTopic(this.topic.attributes.type),
+                    canJoinTeam = TeamUtils.canJoinTeam(this.context.userInfo, this.topic.attributes.type),
```

Comment:
Why are we completely removing the `!isInstructorManagedTopic`?

Reply:
Because that variable can't define if a learner can join a team, and it was used here altough it's not it's purpose.

An open managed team, can be created only by staff, but students can join/leave freely

Reply:
We should not modify an existing behavior; we should only add a new one without affecting what's already here to avoid breaking things. Can we ask the maintainer team about this change?

Reply:
This doesn't modify the existing behavior. In this context, this variable is only used to verify if the leave team button should appear

Reply:
Still I still don't get why we're modifying this line instead of adding a new check like `|| canJoinTeam` so we have fewer changes which implies less impact

Reply:
Because canJoinTeam is a superset of isInstructorManagedTeam, but I can do the refactor